src/s3fs.cpp

```diff
@@ -393,78 +402,24 @@ static int get_object_attribute(const char* path, struct stat* pstbuf, headers_t
         return -ENOENT;
     }
 
-    // At first, check path
-    strpath     = path;
-    result      = s3fscurl.HeadRequest(strpath.c_str(), (*pheader));
-    s3fscurl.DestroyCurlHandle();
-
-    // if not found target path object, do over checking
-    if(-EPERM == result){
-        // [NOTE]
-        // In case of a permission error, it exists in directory
-        // file list but inaccessible. So there is a problem that
-        // it will send a HEAD request every time, because it is
-        // not registered in the Stats cache.
-        // Therefore, even if the file has a permission error, it
-        // should be registered in the Stats cache. However, if
-        // the response without modifiying is registered in the
-        // cache, the file permission will be 0644(umask dependent)
-        // because the meta header does not exist.
-        // Thus, set the mode of 0000 here in the meta header so
-        // that s3fs can print a permission error when the file
-        // is actually accessed.
-        // It is better not to set meta header other than mode,
-        // so do not do it.
-        //
-        (*pheader)["x-amz-meta-mode"] = str(0);
```

Comment:
Is it possible to confirm in the same situation as this?

If there is an object that causes an access permission error (returned from the S3 server side) and the cache is not registered, access to that object will send a HEAD request every time.
Maybe it should be handled by the get_object_attribute_fill_dir function.